utils/stringforwarder/stringforwarder.go

```diff
@@ -7,59 +7,70 @@ import (
 	"sync/atomic"
 )
 
-// stringForwarder will take messages on a receive() method and forward them to
+// StringForwarder will take messages on a receive() method and forward them to
 // callback, but will drop them if callback() has not finished processing the
 // last message. We will track how many messages we have discarded.
-type stringForwarder struct {
-	callback     func(string)
-	buffer       chan string
-	stopch       chan struct{}
-	started      chan struct{}
+type StringForwarder struct {
+	messages     chan<- string
+	done         chan<- struct{}
 	discardCount uint64
```

Comment:
Minor: Why export the type now?  Why not the fields too?  TBH, neither really seem important enough to be part of this patch.